ui-tests/src/test/java/io/syndesis/qe/steps/customizations/api_client_connectors/ApiClientConnectorsSteps.java

```diff
@@ -0,0 +1,180 @@
+		apiClientConnectorsPage.validate();
+	}
+
+	@And("^(\\w+) navigates to the next Api Connector wizard step \"([^\"]*)\"$")
+	public void she_navigates_to_the_next_Api_Connector_wizard_step(String user, String step) {
+		wizard.nextStep();
+
+		//validation ensures that multiple clicking the Next button doesn't click on the same button multiple times before the next step loads and displays
+		wizard.getCurrentStep().validate();
+	}
+
+	@And("^(\\w+) creates new connector$")
+	public void she_creates_new_connector(String user) {
+		wizard.nextStep();
+	}
+
+	@Then("^(\\w+) opens new Api Connector wizard$")
+	public void she_opens_new_api_connector_wizard(String user) throws Throwable {
+		apiClientConnectorsPage.startWizard();
+		wizard = new ApiClientConnectorWizard();
+	}
+
+	@Then("^(\\w+) uploads swagger file$")
+	public void she_uploads_swagger_file(String user, DataTable fileParams) throws Throwable {
+		uploadSwaggerWizardStep.validate();
+		List<List<String>> dataRows = fileParams.cells(0);
+		List<String> sourceTypes = new ArrayList<String>();
+		List<String> urls = new ArrayList<String>();
+
+		for (List<String> row : dataRows) {
+			sourceTypes.add(row.get(0));
+			urls.add(row.get(1));
+		}
+
+		uploadSwaggerWizardStep.upload(sourceTypes.get(0), urls.get(0));
+	}
+
+	@Then("^(\\w+) reviews swagger actions$")
+	public void she_reviews_swagger_actions(String user) throws Throwable {
+		reviewSwaggerActionsWizardStep.validate();
+		reviewSwaggerActionsWizardStep.review();
+		reviewSwaggerActionsWizardStep.nextWizardStep();
+	}
+
+	@Then("(\\w+) sets up security by \"([^\"]*)\"$")
+	public void she_sets_up_security_by(String user, String authType) throws Throwable {
+		securityWizardStep.validate();
+		//String authType, String authenticationUrl, String authorizationUrl, String scopes
+		switch (authType) {
+			case "OAuth 2.0":
+				securityWizardStep.setUpSecurity(TestConfiguration.syndesisUrl() + TestConfiguration.syndesisCallbackUrlSuffix());
+				break;
+			default:
+				throw new UnsupportedOperationException("The Auth type < " + authType + "> is not implemented by the test.");
+		}
+	}
+
+	@Then("^(\\w+) reviews general info$")
+	public void she_reviews_general_info(String user) throws Throwable {
+		generalConnectorInfoWizardStep.validate();
+		generalConnectorInfoWizardStep.review();
+		generalConnectorInfoWizardStep.validate();
+	}
+
+	@Then("(\\w+) is presented with the new connector \"([^\"]*)\"$")
+	public void she_is_presented_with_the_new_connector(String user, String connectorName) throws Throwable {
+		Assert.assertTrue("Connector [" + connectorName + "] is not listed.", apiClientConnectorsPage.isConnectorPresent(connectorName));
+	}
+
+	@Then("^(\\w+) creates new API connector \"([^\"]*)\"$")
+	public void she_creates_new_api_connector(String user, String connectorName, DataTable properties) throws Throwable {
+
+		String securityAuthType = null;
+
+		for (List<String> property : properties.raw()) {
```

Comment:
This construction seems pretty complex and very understandable at a first glance. How it will work for Basic Auth definition?

Reply:
others than OAuth has not been implemented yet. It will be added in future as we have different swagger test files. I cant see how exactly the UI looks with different UI. But in general...Security page object will have methods for other security options.
The scenario step will get DataTable for other security options that are not needed now. It is easily extensible

This is metastep that will need potentially many arguments from datatable because there will be different list of different attributes for each potential custom connector variant. It will get extended as the new configurations are implemented.